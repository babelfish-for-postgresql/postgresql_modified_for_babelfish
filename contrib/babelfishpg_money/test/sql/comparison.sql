-- True comparisons

SELECT '123'::FIXEDDECIMAL < '123.01'::FIXEDDECIMAL;

SELECT '123'::FIXEDDECIMAL <= '123.01'::FIXEDDECIMAL;

SELECT '123'::FIXEDDECIMAL > '122.99'::FIXEDDECIMAL;

SELECT '123'::FIXEDDECIMAL >= '122.99'::FIXEDDECIMAL;

SELECT '123.00'::FIXEDDECIMAL = '123'::FIXEDDECIMAL;

-- Compare to int4

SELECT '123'::INT < '123.01'::FIXEDDECIMAL;

SELECT '123'::INT <= '123.01'::FIXEDDECIMAL;

SELECT '123'::INT > '122.99'::FIXEDDECIMAL;

SELECT '123'::INT >= '122.99'::FIXEDDECIMAL;

SELECT '123'::INT = '123.00'::FIXEDDECIMAL;

-- Compare to int4 reversed

SELECT '123.01'::FIXEDDECIMAL > '123'::INT;

SELECT  '123.01'::FIXEDDECIMAL >= '123'::INT;

SELECT '122.99'::FIXEDDECIMAL < '123'::INT;

SELECT '122.99'::FIXEDDECIMAL <= '123'::INT;

SELECT '123.00'::FIXEDDECIMAL = '123'::INT;

-- Compare to int2

SELECT '123'::SMALLINT < '123.01'::FIXEDDECIMAL;

SELECT '123'::SMALLINT <= '123.01'::FIXEDDECIMAL;

SELECT '123'::SMALLINT > '122.99'::FIXEDDECIMAL;

SELECT '123'::SMALLINT >= '122.99'::FIXEDDECIMAL;

SELECT '123'::SMALLINT = '123.00'::FIXEDDECIMAL;

-- Compare to int4 reversed

SELECT '123.01'::FIXEDDECIMAL > '123'::SMALLINT;

SELECT  '123.01'::FIXEDDECIMAL >= '123'::SMALLINT;

SELECT '122.99'::FIXEDDECIMAL < '123'::SMALLINT;

SELECT '122.99'::FIXEDDECIMAL <= '123'::SMALLINT;

SELECT '123.00'::FIXEDDECIMAL = '123'::SMALLINT;

-- False comparisons
SELECT '123'::FIXEDDECIMAL >= '123.01'::FIXEDDECIMAL;

SELECT '123'::FIXEDDECIMAL > '123.01'::FIXEDDECIMAL;

SELECT '123'::FIXEDDECIMAL <= '122.99'::FIXEDDECIMAL;

SELECT '123'::FIXEDDECIMAL < '122.99'::FIXEDDECIMAL;

SELECT '123.00'::FIXEDDECIMAL <> '123'::FIXEDDECIMAL;

-- Compare to int4

SELECT '123'::INT >= '123.01'::FIXEDDECIMAL;

SELECT '123'::INT > '123.01'::FIXEDDECIMAL;

SELECT '123'::INT <= '122.99'::FIXEDDECIMAL;

SELECT '123'::INT < '122.99'::FIXEDDECIMAL;

SELECT '123'::INT <> '123.00'::FIXEDDECIMAL;

-- Compare to int4 reversed

SELECT '123.01'::FIXEDDECIMAL <= '123'::INT;

SELECT  '123.01'::FIXEDDECIMAL < '123'::INT;

SELECT '122.99'::FIXEDDECIMAL >= '123'::INT;

SELECT '122.99'::FIXEDDECIMAL > '123'::INT;

SELECT '123.00'::FIXEDDECIMAL <> '123'::INT;

-- Compare to int2

SELECT '123'::SMALLINT >= '123.01'::FIXEDDECIMAL;

SELECT '123'::SMALLINT > '123.01'::FIXEDDECIMAL;

SELECT '123'::SMALLINT <= '122.99'::FIXEDDECIMAL;

SELECT '123'::SMALLINT < '122.99'::FIXEDDECIMAL;

SELECT '123'::SMALLINT <> '123.00'::FIXEDDECIMAL;

-- Compare to int4 reversed

SELECT '123.01'::FIXEDDECIMAL <= '123'::SMALLINT;

SELECT  '123.01'::FIXEDDECIMAL < '123'::SMALLINT;

SELECT '122.99'::FIXEDDECIMAL >= '123'::SMALLINT;

SELECT '122.99'::FIXEDDECIMAL > '123'::SMALLINT;

SELECT '123.00'::FIXEDDECIMAL <> '123'::SMALLINT;
