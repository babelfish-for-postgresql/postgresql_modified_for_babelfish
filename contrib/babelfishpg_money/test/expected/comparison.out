-- True comparisons
SELECT '123'::FIXEDDECIMAL < '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::FIXEDDECIMAL <= '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::FIXEDDECIMAL > '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::FIXEDDECIMAL >= '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123.00'::FIXEDDECIMAL = '123'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

-- Compare to int4
SELECT '123'::INT < '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::INT <= '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::INT > '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::INT >= '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::INT = '123.00'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

-- Compare to int4 reversed
SELECT '123.01'::FIXEDDECIMAL > '123'::INT;
 ?column? 
----------
 t
(1 row)

SELECT  '123.01'::FIXEDDECIMAL >= '123'::INT;
 ?column? 
----------
 t
(1 row)

SELECT '122.99'::FIXEDDECIMAL < '123'::INT;
 ?column? 
----------
 t
(1 row)

SELECT '122.99'::FIXEDDECIMAL <= '123'::INT;
 ?column? 
----------
 t
(1 row)

SELECT '123.00'::FIXEDDECIMAL = '123'::INT;
 ?column? 
----------
 t
(1 row)

-- Compare to int2
SELECT '123'::SMALLINT < '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::SMALLINT <= '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::SMALLINT > '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::SMALLINT >= '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

SELECT '123'::SMALLINT = '123.00'::FIXEDDECIMAL;
 ?column? 
----------
 t
(1 row)

-- Compare to int4 reversed
SELECT '123.01'::FIXEDDECIMAL > '123'::SMALLINT;
 ?column? 
----------
 t
(1 row)

SELECT  '123.01'::FIXEDDECIMAL >= '123'::SMALLINT;
 ?column? 
----------
 t
(1 row)

SELECT '122.99'::FIXEDDECIMAL < '123'::SMALLINT;
 ?column? 
----------
 t
(1 row)

SELECT '122.99'::FIXEDDECIMAL <= '123'::SMALLINT;
 ?column? 
----------
 t
(1 row)

SELECT '123.00'::FIXEDDECIMAL = '123'::SMALLINT;
 ?column? 
----------
 t
(1 row)

-- False comparisons
SELECT '123'::FIXEDDECIMAL >= '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::FIXEDDECIMAL > '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::FIXEDDECIMAL <= '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::FIXEDDECIMAL < '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123.00'::FIXEDDECIMAL <> '123'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

-- Compare to int4
SELECT '123'::INT >= '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::INT > '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::INT <= '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::INT < '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::INT <> '123.00'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

-- Compare to int4 reversed
SELECT '123.01'::FIXEDDECIMAL <= '123'::INT;
 ?column? 
----------
 f
(1 row)

SELECT  '123.01'::FIXEDDECIMAL < '123'::INT;
 ?column? 
----------
 f
(1 row)

SELECT '122.99'::FIXEDDECIMAL >= '123'::INT;
 ?column? 
----------
 f
(1 row)

SELECT '122.99'::FIXEDDECIMAL > '123'::INT;
 ?column? 
----------
 f
(1 row)

SELECT '123.00'::FIXEDDECIMAL <> '123'::INT;
 ?column? 
----------
 f
(1 row)

-- Compare to int2
SELECT '123'::SMALLINT >= '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::SMALLINT > '123.01'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::SMALLINT <= '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::SMALLINT < '122.99'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

SELECT '123'::SMALLINT <> '123.00'::FIXEDDECIMAL;
 ?column? 
----------
 f
(1 row)

-- Compare to int4 reversed
SELECT '123.01'::FIXEDDECIMAL <= '123'::SMALLINT;
 ?column? 
----------
 f
(1 row)

SELECT  '123.01'::FIXEDDECIMAL < '123'::SMALLINT;
 ?column? 
----------
 f
(1 row)

SELECT '122.99'::FIXEDDECIMAL >= '123'::SMALLINT;
 ?column? 
----------
 f
(1 row)

SELECT '122.99'::FIXEDDECIMAL > '123'::SMALLINT;
 ?column? 
----------
 f
(1 row)

SELECT '123.00'::FIXEDDECIMAL <> '123'::SMALLINT;
 ?column? 
----------
 f
(1 row)

